Added Gas Data to Dashboard

benediktkaiser opened the pull request:
closes #198 

![Screenshot 2022-04-13 at 13 18 22](https://user-images.githubusercontent.com/30908158/163169277-5d61fbfc-8e64-44ce-bc7b-d554f52ed529.png)

In this issue it was stated that the decimals should be as exact as possible. To ensure this I used our `FormatCurrency` component, which had a built in formatting we agreed on for the rest of the UI. Is this formatting fine, or should I always show the exact number, no matter how long it is?

benediktkaiser commented on frontend/components/DashboardAuctionsView.vue:
```diff
@@ -18,7 +18,7 @@
         </div>
         <div class="Block max-w-screen-lg">
-            <CollateralTable :collaterals="collaterals" class="overflow-x-scroll" />
+            <CollateralTable :collaterals="collaterals" class="overflow-x-auto" />
```
I also fixed a small issue here with the scrollbar. Before the scrollbar was always shown, even if it was not needed. With this new class it only shows once it is actually needed.

benediktkaiser commented:
> `getGasParameters` is in WEI

That makes a lot of sense. I was a bit sceptical myself haha.

valiafetisov commented:
- [ ] The third column contains value in DAI, as specified in the title, so no need to add more than 2 decimals
- [ ] First 4 rows will contain very small values and using mathmatical notation is not acceptable as suggested, please use enough decimal points or format it to GWEI by shifting it according to the table https://user-images.githubusercontent.com/9872658/34238788-2f7fa014-e636-11e7-8f62-cae47040f247.png by 9 decimal points

benediktkaiser commented:
Does this look correct?
![Screenshot 2022-04-13 at 16 24 36](https://user-images.githubusercontent.com/30908158/163202508-8e725d73-6855-449c-9e96-95a203cbcb49.png)

valiafetisov commented:
Can I also ask you to add restart transaction fee in DAI? I just checked that it's not in the types, but you can quickly add it to the `getApproximateTransactionFees`.

benediktkaiser commented:
Added `RestartFeeDAI` and corrected all the formatting! 

![Screenshot 2022-04-13 at 16 48 02](https://user-images.githubusercontent.com/30908158/163207418-5d2c4dce-7fe4-44d4-9daa-111c8e0baf43.png)

valiafetisov commented:
`maxPriorityFeePerGas` suppose to be 1.5 Gwei according to this line: https://github.com/sidestream-tech/unified-auctions-ui/blob/493dcd57d7e7b43b9d0a47183e99e0be27230a75/core/src/gas.ts#L12 Why is it displayed as 1?

benediktkaiser commented:
> `maxPriorityFeePerGas` suppose to be 1.5 Gwei according to this line:
> 
> https://github.com/sidestream-tech/unified-auctions-ui/blob/493dcd57d7e7b43b9d0a47183e99e0be27230a75/core/src/gas.ts#L12
> 
> Why is it displayed as 1 now?

It was because I was using `pharseInt()`. I removed this and it all looks good now!

![Screenshot 2022-04-13 at 17 05 20](https://user-images.githubusercontent.com/30908158/163211116-07422275-734a-4b5d-82b2-3b75bf5dc05c.png)

valiafetisov commented on frontend/components/GasTable.vue:
```diff
@@ -44,11 +38,9 @@
                     <td class="Body">
-                        <FormatCurrency
-                            v-if="gasParameters && gasParameters.maxFeePerGas"
-                            :value="parseInt(gasParameters.maxFeePerGas)"
```
Please convert it to BigNumber, not number and let's still use `FormatCurrency` since it's animated (:

valiafetisov reviewed (commented):
Do we also periodically refetch those values? Once in 30 seconds will be enough here.

valiafetisov commented on frontend/store/gas.ts:
```diff
@@ -0,0 +1,102 @@
+        const gasParameters: GasParameters = {
+            maxFeePerGas:
+                initialGasParameters.maxFeePerGas &&
+                new BigNumber(initialGasParameters.maxFeePerGas)
+                    .shiftedBy(-ETH_NUMBER_OF_DIGITS)
+                    .shiftedBy(GWEI_NUMBER_OF_DIGITS)
+                    .toString(),
```
As discussed, please never use `.toString()` on big numbers. Also, my suggestion would be of course to format numbers in the view, not in the store

benediktkaiser replied:
I moved the formatting to the component, bypassing all this `toString` back and forth.

benediktkaiser commented:
> Do we also periodically refetch those values? Once in 30 seconds will be enough here.

I missed this. I can add it right now.

benediktkaiser commented on frontend/components/GasTable.vue:
```diff
@@ -150,6 +159,11 @@ export default Vue.extend({
     },
+    methods: {
+        formatGasParameters(value): BigNumber {
+            return new BigNumber(value).shiftedBy(-ETH_NUMBER_OF_DIGITS).shiftedBy(GWEI_NUMBER_OF_DIGITS);
```
I committed the change!

LukSteib reviewed (changes requested):
- [ ] The `Gas Price` in row one is continously UNKNOWN. Do we need it in the first place? AFAIK we use it for `Max Fee Per Gas`, so IMO we can also get rid of the `Gas Price` here
- [ ] We might rename the header row of the second column from `Value in ETH` to `Value` only since we display a mix of ETH and GWEI

benediktkaiser commented:
> * [ ]  The `Gas Price` in row one is continously UNKNOWN. Do we need it in the first place? AFAIK we use it for `Max Fee Per Gas`, so IMO we can also get rid of the `Gas Price` here

> gasPrice or maxPriorityFeePerGas and maxFeePerGas depending on the network some are expected to be undefined

I think @valiafetisov mentioned that not all of the three values will be found on every network.

valiafetisov commented:
One other neat peak: can we please format all units the same way? I suppose, uppercase?
> <img width="96" alt="Screenshot 2022-04-14 at 09 29 05" src="https://user-images.githubusercontent.com/3393626/163335809-1ec391b2-fba3-4eaa-9c2b-20f4cc9dc0f1.png">

valiafetisov commented:
Approved, btw
